### ISSUE TITLE
Error in operator reshape 

### ISSUE THREAD

**Issue by hsddlz** [apache/incubator-mxnet#10065]
when I bind the data .It told me like this:

simple_bind error. Arguments:
data: (32, 1L, 32L, 286L)
softmax_label: (32, 21L)
Error in operator reshape0: [19:25:11] src/operator/tensor/./matrix_op-inl.h:157: Check failed: oshape.Size() == dshape.Size() (17920 vs. 573440) Target shape size is different to source. Target: [35,512]
Source: [32,128,4,35]

my ndarray data shape is [DataDesc[data,(32, 1L, 32L, 286L),<type 'numpy.float32'>,NCHW]]

this is my code 
`stride=(1,1)
bn_mom=0.9
fix_gamma=False
eps=2e-5
rnn_length =35
rnn_dimen=512
rnn_size = 256

x = mx.sym.Variable("data")
label = mx.sym.Variable('label')
for i, n_cnn in enumerate([3, 4, 6]):
    for j in range(n_cnn):
        x = mx.symbol.Convolution(data=x, num_filter=32*2**i, kernel=(3, 3), stride=stride, pad=(1,1))
        x = mx.symbol.BatchNorm(data=x, fix_gamma=fix_gamma, eps=eps, momentum=bn_mom)
        x = mx.symbol.Activation(data=x, act_type='relu')
    
    x = mx.sym.Pooling(data=x, kernel=(2, 2), stride=(2, 2), pool_type='max')
x=mx.symbol.reshape(x,shape=(rnn_length, rnn_dimen))`

how can I fix it.thx

**Comment by hsddlz** [comment]
shape=(batch_size,1,rnn_length, rnn_dimen)    sorry
